app/scripts/controllers/main.js

```diff
@@ -8,24 +8,64 @@
  * Controller of the moodCatApp
  */
 angular.module('moodCatApp')
-  .controller('MainCtrl', function ($scope, $timeout, soundCloudService) {
+  .service('roomService', function($http) {
+    this.fetchRooms = function() {
+      return $http.get('/api/rooms/');
+      //return $http.get('/mocks/rooms.json');
+    }
+  })
+  .service('chatService', function($http) {
+    this.sendChatMessage = function(mess) {
+
+      var request = $http.post('/api/rooms/1/chat/post', angular.toJson(mess));
+      //
+      // // Store the data-dump of the FORM scope.
+      request.success(
+        function(response) {
+          console.log(angular.toJson(response));
+
+        }
+      );
+      request.error(
+        function() {
+          alert("Failed to send message!");
```

Comment:
Please use the `$log` service for logging. Console breaks on some browsers and alert is not that friendly. `$log` has methods for logging at various levels: `$log.warn` and `$log.info`.

You can also inline this:

```js
request.error($log.warn.bind($log, "Failed to send message!"));
``